### ISSUE TITLE
linter: false positive in jest/no-standalone-expect

### ISSUE THREAD

**Issue by jackstevenson** [oxc-project/oxc#21897]
### What version of Oxlint are you using?

1.62.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["jest"],
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "jest/no-standalone-expect": "error"
  }
}
```

### What happened?

We have a number of tests where the assertion is defined in a helper function and Oxlint is flagging it as an error.

```js
const assertFn = (a, b) => {
  expect(a).toBe(b); // No error
};

const assertions = {
  assertFn: (a, b) => {
    expect(a).toBe(b); // Oxc Error: `expect` must be inside of a test block.
  }
};

test('test', () => {
  assertFn(1, 1);
  assertions.assertFn(1, 1);
});
```

[Playground repro](https://playground.oxc.rs/?lintRules=jest%2Fno-standalone-expect&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=const+assertFn+%3D+%28a%2C+b%29+%3D%3E+%7B%0A++expect%28a%29.toBe%28b%29%3B+%2F%2F+No+error%0A%7D%3B%0A%0Aconst+assertions+%3D+%7B%0A++assertFn%3A+%28a%2C+b%29+%3D%3E+%7B%0A++++expect%28a%29.toBe%28b%29%3B+%2F%2F+Oxc+Error%3A+%60expect%60+must+be+inside+of+a+test+block.%0A++%7D%0A%7D%3B%0A%0Atest%28%27test%27%2C+%28%29+%3D%3E+%7B%0A++assertFn%281%2C+1%29%3B%0A++assertions.assertFn%281%2C+1%29%3B%0A%7D%29%3B%0A)
